agent/consul/multilimiter/multilimiter.go

```diff
@@ -0,0 +1,128 @@
+package multilimiter
+
+import (
+	"context"
+	radix "github.com/hashicorp/go-immutable-radix"
+	"golang.org/x/time/rate"
+	"sync/atomic"
+	"time"
+)
+
+type LimitedEntity interface {
```

Comment:
Nice. We can iterate/add properties to this in later phases.  Then in the case of limiting on IP, tenant, or service, we could add properties for things like LimitType (IP limiting vs tenant limiting vs service limiting vs global limiting) and Scope (IP1 vs IP2 or Tenant1 vs Tenant2).  So, basically we don't need to figure out that structure entirely yet and we can iterate this interface without changing the signature of MultiLimiter's `Allow()`?